subsys/logging/log_backend_net.c

```diff
@@ -0,0 +1,238 @@
+			 ROUND_UP(CONFIG_LOG_BACKEND_NET_MAX_BUF_SIZE /
+				  CONFIG_NET_BUF_DATA_SIZE, 1) *
+			 CONFIG_LOG_BACKEND_NET_MAX_BUF);
+
+static struct k_mem_slab *get_tx_slab(void)
+{
+	return &syslog_tx_pkts;
+}
+
+struct net_buf_pool *get_data_pool(void)
+{
+	return &syslog_tx_bufs;
+}
+
+static int line_out(u8_t *data, size_t length, void *output_ctx)
+{
+	struct net_context *ctx = (struct net_context *)output_ctx;
+	int ret = -ENOMEM;
+	struct net_pkt *pkt;
+
+	if (!ctx) {
+		return length;
+	}
+
+	pkt = net_pkt_get_tx(ctx, K_NO_WAIT);
+	if (!pkt) {
+		goto fail;
+	}
+
+	if (net_pkt_append_all(pkt, length, data, K_NO_WAIT) == false) {
+		goto fail;
+	}
+
+	ret = net_context_send(pkt, NULL, K_NO_WAIT, NULL, NULL);
+	if (ret < 0) {
+		goto fail;
+	}
+
+	DBG(data);
+
+fail:
+	if (ret < 0 && pkt) {
+		net_pkt_unref(pkt);
+	}
+
+	return length;
+}
+
+LOG_OUTPUT_DEFINE(log_output, line_out, output_buf, sizeof(output_buf));
+
+static int do_net_init(void)
+{
+	struct sockaddr *local_addr = NULL;
+	struct sockaddr_in6 local_addr6;
+	struct sockaddr_in local_addr4;
+	socklen_t server_addr_len;
+	struct net_context *ctx;
+	int ret;
+
+	if (IS_ENABLED(CONFIG_NET_IPV4) && server_addr.sa_family == AF_INET) {
+		local_addr = (struct sockaddr *)&local_addr4;
+		server_addr_len = sizeof(struct sockaddr_in);
+		local_addr4.sin_port = 0;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_IPV6) && server_addr.sa_family == AF_INET6) {
+		local_addr = (struct sockaddr *)&local_addr6;
+		server_addr_len = sizeof(struct sockaddr_in6);
+		local_addr6.sin6_port = 0;
+	}
+
+	if (!local_addr) {
+		DBG("Server address unknown\n");
+		return -EINVAL;
+	}
+
+	local_addr->sa_family = server_addr.sa_family;
+
+	ret = net_context_get(server_addr.sa_family, SOCK_DGRAM, IPPROTO_UDP,
+			      &ctx);
+	if (ret < 0) {
+		DBG("Cannot get context (%d)\n", ret);
+		return ret;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_HOSTNAME_ENABLE)) {
+		memcpy(hostname, net_hostname_get(), MAX_HOSTNAME_LEN);
+
+	} else if (IS_ENABLED(CONFIG_NET_IPV6) &&
+		   server_addr.sa_family == AF_INET6) {
+		const struct in6_addr *src;
+
+		src = net_if_ipv6_select_src_addr(
+			NULL, &net_sin6(&server_addr)->sin6_addr);
+		if (src) {
+			net_addr_ntop(AF_INET6, src, hostname,
+				      MAX_HOSTNAME_LEN);
+
+			net_ipaddr_copy(&local_addr6.sin6_addr, src);
+		} else {
+			goto unknown;
+		}
+
+	} else if (IS_ENABLED(CONFIG_NET_IPV4) &&
+		   server_addr.sa_family == AF_INET) {
+		struct net_if_ipv4 *ipv4;
+		struct net_if *iface;
+
+		iface = net_if_ipv4_select_src_iface(
+					&net_sin(&server_addr)->sin_addr);
+		ipv4 = iface->config.ip.ipv4;
+
+		net_ipaddr_copy(&local_addr4.sin_addr,
+				&ipv4->unicast[0].address.in_addr);
+
+		net_addr_ntop(AF_INET, &local_addr4.sin_addr, hostname,
+			      MAX_HOSTNAME_LEN);
+	} else {
+	unknown:
+		DBG("Cannot setup local context\n");
+		return -EINVAL;
+	}
+
+	ret = net_context_bind(ctx, local_addr, server_addr_len);
+	if (ret < 0) {
+		DBG("Cannot bind context (%d)\n", ret);
```

Comment:
you are mising LOG_ERR usage and DBG in this file. I would stick to LOG_ERR.

Reply:
The idea with DBG() was to just use printk() and not any of the LOG_*() macros so that we would not get circular prints inside the logging backend.